librz/include/rz_search.h

```diff
@@ -51,39 +51,62 @@ typedef struct rz_search_hit_t {
 
 typedef int (*RzSearchCallback)(RzSearchKeyword *kw, void *user, ut64 where);
 
-typedef struct rz_search_t {
-	int n_kws; // hit${n_kws}_${count}
-	int mode;
+struct rz_search_t;
+
+typedef struct rz_search_params_t {
+	RzList *boundaries;
+	const char *cmd_hit; ///< cmd.hit
+	RzInterval itv;
 	ut32 pattern_size;
-	ut32 string_min; // max length of strings for RZ_SEARCH_STRING
-	ut32 string_max; // min length of strings for RZ_SEARCH_STRING
+	RzSearchMode mode;
+
+	bool aes_search;
+	bool privkey_search;
+	bool inverse;
+	bool backwards;
+
+	int search_align; ///< search.align
+	bool search_contiguous; ///< search.contiguous
+	int search_distance; ///< search.distance
+	bool search_flags; ///< search.flags
+	ut64 search_from; ///< search.from
+	ut64 search_to; ///< search.to
+	ut64 search_maxhits; ///< search.maxhits
+	ut32 search_maxlength; ///< search.maxlength: max length of strings for RZ_SEARCH_STRING
+	ut32 search_minlength; ///< search.minlength: min length of strings for RZ_SEARCH_STRING
+	bool search_overlap; ///< search.overlap: whether two matches can overlap
+	const char *search_prefix; ///< search.prefix
+	bool search_show; ///< search.show
+
+	RzList *kws; // TODO: Use rz_search_kw_new ()
+	int n_kws; // hit${n_kws}_${count}
+	RzIOBind iob;
+	int (*update)(struct rz_search_t *s, ut64 from, const ut8 *buf, int len);
+} RzSearchParams;
+
+typedef struct rz_search_t {
 	void *data; // data used by search algorithm
 	void *user; // user data passed to callback
 	RzSearchCallback callback;
 	ut64 nhits;
-	ut64 maxhits; // search.maxhits
-	RzList *hits;
-	int distance;
-	int inverse;
-	bool overlap; // whether two matches can overlap
-	int contiguous;
-	int align;
-	int (*update)(struct rz_search_t *s, ut64 from, const ut8 *buf, int len);
-	RzList *kws; // TODO: Use rz_search_kw_new ()
-	RzIOBind iob;
-	char bckwrds;
+	RzList /*<RzSearchHit>*/ *hits;
+	RzSearchParams *params;
 } RzSearch;
 
+typedef struct rz_core_t RzCore;
```

Comment:
This is not good here. There should be no reference to core

Reply:
Yeah, it's no longer necessary. I'll remove it.